server/src/test/java/org/elasticsearch/action/bulk/BulkIntegrationIT.java

```diff
@@ -40,4 +48,51 @@ public void testBulkIndexCreatesMapping() throws Exception {
             assertTrue(mappingsResponse.getMappings().get("logstash-2014.03.30").containsKey("logs"));
         });
     }
+
+    /**
+     * This tests that the {@link TransportBulkAction} evaluates alias routing values correctly when dealing with
+     * an alias pointing to multiple indices, while a write index exits.
+     */
+    public void testBulkWithWriteIndexAndRouting() {
+        Map<String, Integer> twoShardsSettings = Collections.singletonMap(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 2);
+        client().admin().indices().prepareCreate("index1")
+            .addAlias(new Alias("alias1").indexRouting("0")).setSettings(twoShardsSettings).get();
+        client().admin().indices().prepareCreate("index2")
+            .addAlias(new Alias("alias1").indexRouting("0").writeIndex(randomFrom(false, null)))
+            .setSettings(twoShardsSettings).get();
+        client().admin().indices().prepareCreate("index3")
+            .addAlias(new Alias("alias1").indexRouting("1").writeIndex(true)).setSettings(twoShardsSettings).get();
+
+        IndexRequest indexRequest = new IndexRequest("index3", "type", "id");
+        boolean indexWithRouting = randomBoolean();
+        if (indexWithRouting) {
+            indexRequest.routing("0");
+        }
+        indexRequest.source(Collections.singletonMap("foo", "bar"));
+        IndexRequest indexRequestWithAlias = new IndexRequest("alias1", "type", "id");
+        indexRequestWithAlias.source(Collections.singletonMap("foo", "baz"));
+        BulkResponse bulkResponse = client().prepareBulk()
+            .add(indexRequest).add(indexRequestWithAlias).get();
+        assertThat(bulkResponse.getItems()[0].getResponse().getIndex(), equalTo("index3"));
+        assertThat(bulkResponse.getItems()[0].getResponse().getShardId().getId(), equalTo(1));
+        assertThat(bulkResponse.getItems()[0].getResponse().getVersion(), equalTo(1L));
+        assertThat(bulkResponse.getItems()[0].getResponse().status(), equalTo(RestStatus.CREATED));
+        assertThat(bulkResponse.getItems()[1].getResponse().getIndex(), equalTo("index3"));
+        assertThat(bulkResponse.getItems()[1].getResponse().getShardId().getId(), equalTo(0));
+        assertThat(bulkResponse.getItems()[1].getResponse().getVersion(), equalTo(1L));
+        assertThat(bulkResponse.getItems()[1].getResponse().status(), equalTo(RestStatus.CREATED));
+        GetRequestBuilder getBuilder = client().prepareGet("index3", "type", "id");
+        if (indexWithRouting) {
+            getBuilder.setRouting("0");
+        }
+        assertThat(getBuilder.get().getSource().get("foo"), equalTo("bar"));
+        assertThat(client().prepareGet("index3", "type", "id").setRouting("1").get().getSource().get("foo"), equalTo("baz"));
+
+        client().prepareBulk().add(client().prepareUpdate("alias1", "type", "id").setDoc("foo", "updated")).get();
```

Comment:
we should check errors here?

Reply:
will add. didn't think this was necessary since there are get-action calls verifying success of this operation